src/main.tsx

```diff
@@ -6,30 +6,35 @@ import QuizPage from './pages/QuizPage';
 import StatisticsPage from './pages/StatisticsPage';
 import ResultPage from './pages/ResultPage';
 import './main.css';
+import { Provider } from 'react-redux';
+import store from './store';
+import ErrorBoundary from './components/ErrorBoundary';
 
 const router = createBrowserRouter([
-  {
-    path: '*',
-    element: <ErrorPage />,
-  },
   {
     path: '/',
     element: <SettingsPage />,
   },
   {
     path: '/quiz',
-    element: <QuizPage />,
+    element: (
+      <ErrorBoundary>
```

Comment:
if I were you I would wrap the whole app with the ErrorBoundary. However, it's up to you :)